pkg/operator/ceph/object/spec.go

```diff
@@ -130,6 +161,37 @@ func (c *clusterConfig) makeRGWPodSpec(rgwConfig *rgwConfig) (v1.PodTemplateSpec
 	return podTemplateSpec, nil
 }
 
+func (c *clusterConfig) vaultVolumeMount() v1.VolumeMount {
```

Comment:
We have this code in the OSD already, do not duplicate code.

Reply:
@leseb : I found following from osd/spec.go
```go
       if kmsProvider == secrets.TypeVault {
            if c.spec.Security.KeyManagementService.IsTokenAuthEnabled() {
                getKEKFromKMSContainer := c.generateVaultGetKEK(osdProps)

                // Volume mount to store the encrypted key
                _, volMount := c.getEncryptionVolume(osdProps)
                getKEKFromKMSContainer.VolumeMounts = append(getKEKFromKMSContainer.VolumeMounts, volMount)

                // Now let's see if there is a TLS config we need to mount as well
                _, vaultVolMount := kms.VaultVolumeAndMount(c.spec.Security.KeyManagementService.ConnectionDetails)
                getKEKFromKMSContainer.VolumeMounts = append(getKEKFromKMSContainer.VolumeMounts, vaultVolMount)

                // Add the container to the list of containers
                containers = append(containers, getKEKFromKMSContainer)
            }    
        }    
```
If I understand correctly both volume mounts does not mount the `token` from the secret in the container. Are suggesting me to add this functionality to one of those functions?

Reply:
What about this https://github.com/rook/rook/blob/master/pkg/daemon/ceph/osd/kms/volumes.go#L59?

Reply:
@leseb: Sorry I still didn't understand completely, currently `VaultVolumeAndMount()` mount only SSL certificates passed from vault and token is not mounted. Token is passed as env variable for osd AFAIR. Am I missing something?

Reply:
Just call `VaultVolumeAndMount` with an empty map string and fetch the volumeMount only like:

`_, volMount := VaultVolumeAndMount(...`